src/NServiceBus.Core/Sagas/CustomFinderAdapter.cs

```diff
@@ -1,18 +1,19 @@
 namespace NServiceBus
 {
     using System;
+    using System.Threading.Tasks;
     using NServiceBus.ObjectBuilder;
     using NServiceBus.Sagas;
 
     class CustomFinderAdapter<TSagaData,TMessage> : SagaFinder where TSagaData : IContainSagaData
     {
-        internal override IContainSagaData Find(IBuilder builder, SagaFinderDefinition finderDefinition, SagaPersistenceOptions options, object message)
+        internal override async Task<IContainSagaData> Find(IBuilder builder, SagaFinderDefinition finderDefinition, SagaPersistenceOptions options, object message)
         {
             var customFinderType = (Type)finderDefinition.Properties["custom-finder-clr-type"];
 
             var finder = (IFindSagas<TSagaData>.Using<TMessage>)builder.Build(customFinderType);
 
-            return finder.FindBy((TMessage) message, options);
+            return await finder.FindBy((TMessage)message, options).ConfigureAwait(false);
```

Comment:
I don't think it's necessary since it's nothing someone can change by accident, it just won't compile.